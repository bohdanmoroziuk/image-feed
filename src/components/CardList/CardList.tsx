import { FunctionComponent } from 'react';
import { FlatList } from 'react-native';

import Card from 'src/components/Card';
import getImageUriFromId from 'src/utils/getImageUriFromId';

interface Item {
  id: string;
  author: string;
}

export interface CardListProps {
  items: Item[];
}

const keyExtractor = (item: Item) => item.id;

const renderItem = ({ item }: { item: Item }) => (
  <Card
    key={item.id}
    fullName={item.author}
    image={{ uri: getImageUriFromId(item.id) }}
  />
);

export const CardList: FunctionComponent<CardListProps> = ({ items }) => (
  <FlatList
    data={items}
    renderItem={renderItem}
    keyExtractor={keyExtractor}
  />
);