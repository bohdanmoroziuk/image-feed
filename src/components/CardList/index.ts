import { CardList } from './CardList';

export type { CardListProps } from './CardList';
export default CardList;
