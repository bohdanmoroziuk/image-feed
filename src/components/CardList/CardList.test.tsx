import { render } from '@testing-library/react-native';

import { CardList } from './CardList';

const mockedItems = [
  { id: '0', author: "Bob Ross" },
  { id: '1', author: "Chuck Norris" },
];

describe('CardList', () => {
  it('renders successfully', () => {
    const { toJSON } = render(
      <CardList
        items={mockedItems}
        comments={{}}
        onPressComments={() => {}}
      />
    );

    expect(toJSON()).toMatchSnapshot()
  });

  it('renders correctly', () => {
    const { getByText } = render(
      <CardList
        items={mockedItems}
        comments={{}}
        onPressComments={() => {}}
      />
    );

    expect(getByText(mockedItems[0].author)).toBeDefined();
    expect(getByText(mockedItems[1].author)).toBeDefined();
  });
});