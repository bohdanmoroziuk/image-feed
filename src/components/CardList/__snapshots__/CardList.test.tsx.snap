// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders successfully 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "author": "Bob Ross",
        "id": "0",
      },
      Object {
        "author": "Chuck Norris",
        "id": "1",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 50,
              "paddingHorizontal": 10,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#c0b8e0",
                  "borderRadius": 17.5,
                  "height": 35,
                  "width": 35,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                }
              }
            >
              BR
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "flex": 1,
                "marginHorizontal": 6,
              }
            }
          >
            Bob Ross
          </Text>
        </View>
        <View
          style={
            Object {
              "aspectRatio": 1,
              "backgroundColor": "rgba(0, 0, 0, 0.02)",
            }
          }
        >
          <ActivityIndicator
            size="large"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <Image
            onLoad={[Function]}
            source={
              Object {
                "uri": "https://unsplash.it/600/600?image=0",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 50,
              "paddingHorizontal": 10,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#bb144e",
                  "borderRadius": 17.5,
                  "height": 35,
                  "width": 35,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                }
              }
            >
              CN
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "flex": 1,
                "marginHorizontal": 6,
              }
            }
          >
            Chuck Norris
          </Text>
        </View>
        <View
          style={
            Object {
              "aspectRatio": 1,
              "backgroundColor": "rgba(0, 0, 0, 0.02)",
            }
          }
        >
          <ActivityIndicator
            size="large"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <Image
            onLoad={[Function]}
            source={
              Object {
                "uri": "https://unsplash.it/600/600?image=1",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
